odoo/models.py

```diff
@@ -4843,7 +4854,18 @@ def sudo(self, user=SUPERUSER_ID):
             The returned recordset has the same prefetch object as ``self``.
 
         """
-        return self.with_env(self.env(user=user))
+        if not isinstance(mode, bool):
+            _logger.warning("deprecated use of sudo(user), use with_user(user) instead", stack_info=True)
+            return self.with_user(mode)
+        return self.with_env(self.env(sumode=mode))
+
+    def with_user(self, user):
```

Comment:
maybe `with_user(self, user, mode=False)`?

Reply:
When replacing calls, I have found that separating concerns makes code more readable.
One method, one concern. The only exception being `with_user(u)` that returns records in non-superuser mode.